### PR TITLE
ML-143: Code submission plugin module refactoring

### PR DESCRIPTION
**Pull request by nbirillo**
[ML-143](https://vyahhi.myjetbrains.com/youtrack/issue/ML-143) Code submission clustering / Prepare for integration
- Refactor build.gradle.kts to unify the gradle tasks
- Refactor builders: move the common logic into the base class
- Refactor scripts to run the plugin correctly

### CODE REVIEW

File: code-submissions-clustering-plugin/build.gradle.kts [hunk 1]
```diff
@@ -11,28 +11,31 @@ dependencies {
...
 
-open class BaseCLITask : RunIdeTask() {
+abstract class BaseCLITask : RunIdeTask() {
+    @get:Input
+    abstract val taskName: Property<String>
+
+    private val inputFile: String? by project
+    private val language: String? by project
+    private val outputDir: String? by project
+    private val binaryInput: String? by project
```
Comment (eartser) [inline code-submissions-clustering-plugin/build.gradle.kts:21]: It looks like you have missed a distance limit argument that is only used by `cluster` task

File: code-submissions-clustering-plugin/build.gradle.kts [hunk 2]
```diff
@@ -54,64 +57,27 @@ tasks {
...
+    val calculateDistTaskName = "calculate-dist"
+    register<BaseCLITask>(calculateDistTaskName) {
+        taskName.set(calculateDistTaskName)
+        setArgs()
+        dependsOn(build)
+    }
+
+    val clusterDistTaskName = "cluster"
+    register<BaseCLITask>(clusterDistTaskName) {
+        taskName.set(clusterDistTaskName)
```
Comment (eartser) [inline code-submissions-clustering-plugin/build.gradle.kts:79]: Why not `clusterTaskName`?

File: scripts/src/plugin_runner/utils.py [hunk 3]
```diff
@@ -9,9 +9,13 @@ def configure_parser(parser: ArgumentParser):
         help='Output directory to store folders with output files',
     )
```
Comment (eartser) [inline scripts/src/plugin_runner/utils.py:10]: Should this argument also be named by value of `TaskNamedArgs`?

File: scripts/src/plugin_runner/utils.py [hunk 4]
```diff
@@ -9,9 +9,13 @@ def configure_parser(parser: ArgumentParser):
...
     parser.add_argument(
-        'language',
+        TaskNamedArgs.LANGUAGE.value,
         help='Programming language of code submissions',
     )
+    parser.add_argument(
+        TaskNamedArgs.DISTANCE_LIMIT.value,
+        help='Max distance between two vertices inside one cluster',
+    )
```
Comment (eartser) [inline scripts/src/plugin_runner/utils.py:18]: Do we actually want to consider distance limit as a positional argument for all runners? It is actually used only in `cluster` task

File: scripts/src/utils/runners/abstract_task_runner.py [hunk 5]
```diff
@@ -2,8 +2,25 @@
...
+        **kwargs,
+) -> Dict[TaskNamedArgs, str]:
+    return {
+        TaskNamedArgs.INPUT_FILE:
+            kwargs[input_file_name_in_kwargs](step_id, script_arguments),
+        TaskNamedArgs.OUTPUT_PATH:
+            kwargs['build_output_dir_name'](step_id, script_arguments),
+        TaskNamedArgs.LANGUAGE: script_arguments.language,
+        TaskNamedArgs.DISTANCE_LIMIT: script_arguments.distance_limit,
+    }
```
Comment (eartser) [inline scripts/src/utils/runners/abstract_task_runner.py:23]: Same here: distance limit is not actually a common argument

**Review by eartser — CHANGES_REQUESTED** [review]
Great, thanks! Left a few comments

File: scripts/src/plugin_runner/utils.py (hunk 4 above)
Comment (nbirillo) [reply scripts/src/plugin_runner/utils.py:18]: Could you please fix it in another pr, since we don't have a python runner for this task yet?

Comment (nbirillo) [reply scripts/src/plugin_runner/utils.py:18]: upd: I will redirect your PR to this branch

**Pull request merged**